### ISSUE TITLE
api_downloadDataFile should not catch http errors

### ISSUE THREAD

**Issue by robnagler** [radiasoft/sirepo#6937]
I think this was actually suppressing an error related to a timeout that is now correctly returned as content too large.

**Comment by e-carlin** [comment]
Related https://github.com/radiasoft/sirepo/issues/6812 ?

FWIW this was why it was changed originally https://github.com/radiasoft/sirepo/issues/6021

**Comment by robnagler** [comment]
No, this is a different issue. It was in the [original code](https://github.com/radiasoft/sirepo/blame/4cd3ed38722dfcb4442004435d9164916a7a5163/sirepo/job_api.py#L38). If it is not found, it shouldn't be an [HTTP error coming back from the supervisor](https://github.com/radiasoft/sirepo/blob/8f13a92e041f9c05dc2f1d6e0d2a36b64c1bf96b/sirepo/job_supervisor.py#L777). We can return an error that's interpreted by job_api. HTTP 404 could mean anything.
